packages/@aws-cdk/aws-appmesh/lib/private/utils.ts

```diff
@@ -64,4 +22,14 @@ export interface ConnectionPoolConfig {
    * @default - none
    */
   readonly maxRequests?: number;
+}
+
+/**
+ * Enum of supported AppMesh protocols
+ */
+export enum Protocol {
```

Comment:
So, I moved this to the private section as it was not being used by any exported classes and this was giving a build error.

There is now an issue with the [ECS Service Extensions](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk-containers/ecs-service-extensions/lib/extensions/appmesh.ts#L33) project that was relying on this enum.

Should we move this back to the shared interfaces file and suppress the build warning, or add an enum specific to the ECS Service Extension package?

Reply:
if we added an enum specific to the ECS Service Extension package wouldn't we have to suppress the same warning?

Reply:
Looks like that module is stable. Given that, let's leave this enum as-is, but let's deprecate it. (`@deprecate` JSDoc tag)